File: 2699. Modify Graph Edge Weights.java
class Solution {
    private final static int INF = (int)2e9;
    public int[][] modifiedGraphEdges(int n, int[][] edges, int source, int destination, int target) {
        List<int[]>[] graph = getGraph(n, edges);
        int distance = run(n, graph, source, destination);
        if (distance < target) return new int[0][0];
        boolean matched = distance == target;
        for (int[] edge : edges) {
            if (edge[2] != -1) continue;
            int from = edge[0], to = edge[1];
            edge[2] = matched ? INF : 1;
            graph[from].add(new int[]{to, edge[2]});
            graph[to].add(new int[]{from, edge[2]});
            if (!matched) {
                int newDistance = run(n, graph, source, destination);
                if (newDistance <= target) {
                    matched = true;
                    edge[2] += target - newDistance;
                }
            }
        }
        return matched ? edges : new int[0][0];
    }

    private int run(int n, List<int[]>[] graph, int source, int destination) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b) -> a[1] - b[1]);
        boolean[] visited = new boolean[n];
        pq.add(new int[] {source, 0});
        while (!pq.isEmpty()) {
            int[] node = pq.poll();
            int current = node[0], distance = node[1];
            if (current == destination) return distance;
            if (visited[current]) continue;
            visited[current] = true;
            for (int[] next : graph[current]) {
                int child = next[0], weight = next[1];
                if (visited[child]) continue;
                pq.add(new int[] {child, distance + weight});
            }
        }
        return INF;
    }

    private List<int[]>[] getGraph(int n, int[][] edges) {
        List<int[]>[] graph = new ArrayList[n];
        for (int i = 0; i < n; i++) {
            graph[i] = new ArrayList<>();
        }
        for (int[] edge : edges) {
            int from = edge[0], to = edge[1], weight = edge[2];
            if (weight == -1) continue;
            graph[from].add(new int[]{to, weight});
            graph[to].add(new int[]{from, weight});
        }
        return graph;
    }
}

File: go/2699. Modify Graph Edge Weights.go
func modifiedGraphEdges(n int, edges [][]int, source int, destination int, target int) [][]int {
	graph := getGraph(n, edges)
	distance := run(n, graph, source, destination, target)
	if distance < target {
		return [][]int{}
	}
	matched := distance == target
	for i := 0; i < len(edges); i++ {
		if edges[i][2] != -1 {
			continue
		}
		edges[i][2] = 1
		if matched {
			edges[i][2] += target
		}
		from, to := edges[i][0], edges[i][1]
		graph[from] = append(graph[from], []int{to, edges[i][2]})
		graph[to] = append(graph[to], []int{from, edges[i][2]})
		if !matched {
			newDistance := run(n, graph, source, destination, target)
			if newDistance <= target {
				matched = true
				edges[i][2] += target - newDistance
			}
		}
	}
	if matched {
		return edges
	}
	return [][]int{}
}

func getGraph(n int, edges [][]int) [][][]int {
	graph := make([][][]int, n)
	for i := range graph {
		graph[i] = [][]int{}
	}
	for _, edge := range edges {
		from, to, weight := edge[0], edge[1], edge[2]
		if weight == -1 {
			continue
		}
		graph[from] = append(graph[from], []int{to, weight})
		graph[to] = append(graph[to], []int{from, weight})
	}
	return graph
}

func run(n int, graph [][][]int, source, destination, target int) int {
	pq := &minHeap{}
	heap.Init(pq)
	visited := make([]bool, n)
	heap.Push(pq, []int{source, 0})
	for pq.Len() > 0 {
		node := heap.Pop(pq).([]int)
		current, distance := node[0], node[1]
		if current == destination {
			return distance
		}
		if visited[current] {
			continue
		}
		visited[current] = true
		for _, next := range graph[current] {
			child, weight := next[0], next[1]
			if visited[child] {
				continue
			}
			heap.Push(pq, []int{child, distance + weight})
		}
	}
	return target + 1
}

type minHeap [][]int

func (h minHeap) Len() int            { return len(h) }
func (h minHeap) Less(i, j int) bool  { return h[i][1] < h[j][1] }
func (h minHeap) Swap(i, j int)       { h[i], h[j] = h[j], h[i] }
func (h *minHeap) Push(x interface{}) { *h = append(*h, x.([]int)) }
func (h *minHeap) Pop() interface{} {
	old := *h
	n := len(old)
	x := old[n-1]
	*h = old[0 : n-1]
	return x
}
